portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -130,9 +143,11 @@ private CommentsResult fetchComments(HttpServletRequest request) throws IllegalA
   private Comment transformEntityToComment(Entity entity) {
     String content = (String) entity.getProperty("content");
     String id = (String) entity.getProperty("id");
-    String user = (String) entity.getProperty("user");
+    String userId = (String) entity.getProperty("userId");
+    String userEmail = (String) entity.getProperty("userEmail");
```

Comment:
Instead of storing the email with the comment, it'd be better to get it using the userInfoHelper.
Even better would be to have a getNicknameOrEmailOfUser(userId) function in the userInfoHelper.

Reply:
Thanks for this suggestion. I was thinking about removing user email in comment too, but then I have another two problems I'm not sure how to solve: if I store email with user id in datastore, where should I implement this? Do I store email every time the user login? I'm using username to sort and filter comment entity. If I remove user name in comment, are there other ways to do sorting or filtering by username?

Reply:
For filtering, you should first query the user storage to find out the user id of the user with that email and then use the user id to filter the comments.

Reply:
Okay, get it. Thank you.